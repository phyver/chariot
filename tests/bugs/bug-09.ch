codata stream('x) where
    | Head : stream('x) -> 'x
    | Tail : stream('x) -> stream('x)

:set show_initial_graph true
-- this shouldn't be seen as terminating
val f x = f x.Tail

