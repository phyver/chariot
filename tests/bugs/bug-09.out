>> option verbose is set to 1
>> coinductive type stream was succesfully defined
>> option show_initial_graph is set to true
>> Typing for f successful
>> the definition for f is complete
>> initial callgraph:
>> calls from f to f:
>>     f, x => f, <1ยน>x

!! warning: the definition f is NOT provably total (weight_bound: 2, depth_bound: 2)

